test/functional/readconcern_tests.js

```diff
@@ -1,498 +1,361 @@
+      level: 'majority',
+      commandName: 'find'
+    },
+    {
+      description: 'Should set local readConcern at collection level',
+      level: 'local',
+      commandName: 'find',
+      collectionReadConcernLevel: 'local'
+    },
+    {
+      description: 'Should set majority readConcern at collection level',
+      level: 'majority',
+      commandName: 'find',
+      collectionReadConcernLevel: 'majority'
+    },
+    {
+      description: 'Should set local readConcern using MongoClient',
+      level: 'local',
+      commandName: 'find',
+      urlReadConcernLevel: 'readConcernLevel=local'
+    },
+    {
+      description: 'Should set majority readConcern using MongoClient',
+      level: 'majority',
+      commandName: 'find',
+      urlReadConcernLevel: 'readConcernLevel=majority'
+    },
+    {
+      description: 'Should set majority readConcern using MongoClient with options',
+      level: 'majority',
+      commandName: 'find',
+      urlOptions: 'majority'
+    },
+    {
+      description: 'Should set majority readConcern aggregate command',
+      level: 'majority',
+      commandName: 'aggregate'
+    },
+    {
+      description: 'Should set majority readConcern geoSearch command',
+      level: 'majority',
+      commandName: 'geoSearch'
     }
-  });
-
-  it('Should set majority readConcern using MongoClient', {
-    metadata: { requires: { topology: 'replicaset', mongodb: '>= 3.2' } },
-
-    test: function(done) {
-      var configuration = this.configuration;
-      var listener = require('../..').instrument(function(err) {
-        test.equal(null, err);
-      });
-
-      // Contains all the apm events
-      var started = [];
-      var url = configuration.url();
-      url =
-        url.indexOf('?') !== -1
-          ? f('%s&%s', url, 'readConcernLevel=majority')
-          : f('%s?%s', url, 'readConcernLevel=majority');
-
-      // Connect using mongoclient
-      const client = configuration.newClient(url);
-      client.connect(function(err, client) {
-        expect(err).to.not.exist;
-
-        var db = client.db(configuration.db);
-        test.deepEqual({ level: 'majority' }, db.s.readConcern);
-
-        // Get a collection
-        var collection = db.collection('readConcernCollection');
-        // Validate readConcern
-        test.deepEqual({ level: 'majority' }, collection.s.readConcern);
-        // Perform a find using the readConcern
-        listener.on('started', function(event) {
-          if (event.commandName === 'find') started.push(event);
-        });
-
-        // Execute find
-        collection.find().toArray(function(err) {
-          test.equal(null, err);
-          test.equal(1, started.length);
-          test.deepEqual({ level: 'majority' }, started[0].command.readConcern);
-
-          listener.uninstrument();
-          client.close();
-          done();
-        });
-      });
+  ];
+
+  function run_tests(i, commandName, level) {
```

Comment:
In general, I think it is a little easier to read if the order of these arguments is `commandName, level, index`. That's pretty subjective though so I'm open to it if you think it's cleaner this way!